File: trabalho_do_semestre_java/src/main/java/conexaoSQL/FabricaConexao.java
package conexaoSQL;

import javafx.scene.control.Alert;

import java.sql.*;

/**
 * Classe FabricaConexao responsável por gerenciar a conexao com o banco de dados My SQL.
 */
public class FabricaConexao {

    // atributos necessários para conexao
    private static final String DRIVE = "com.mysql.cj.jdbc.Driver";
    private static final String URL = "jdbc:mysql://localhost:3306/produtos_bd";
    private static final String USER = "Adm";
    private static final String PASS = "Secret_123";

    /**
     * Obtém uma conexao com o banco de dados.
     *
     * @return  uma conexao estabelecida com o banco de dados.
     * @throws RuntimeException quando a conexao nao foi possível com os dados atuais.
     */
    public static Connection getConection(){
        try {
            // busca o driver de conexao
            Class.forName(DRIVE);
            // retorna com os atributos necessários
            return DriverManager.getConnection(URL, USER, PASS);
        } catch (ClassNotFoundException | SQLException ex) {
            throw new RuntimeException("Erro na conexao:", ex);
        }
    }

    /**
     * Fecha a conexão com o banco de dados.
     *
     * @param con a conexão a ser fechada.
     */
    public static void closeConnection(Connection con){

        try {
            if(con != null){
                con.close();
            }
        } catch (SQLException ex) {
            trataProblemaBancoDados(ex);
        }

    }

    /**
     * Fecha a conexão e o PreparedStatement, responsável por modificar o banco de dados
     *
     * @param con a conexão a ser fechada.
     * @param stmt o PreparedStatement a ser fechado.
     */
    public static void closeConnection(Connection con, PreparedStatement stmt){
        // chama o metodo anterior
        closeConnection(con);
        try {
            // se for diferente de nulo, fecha a conexao
            if(stmt != null){
                stmt.close();
            }
        } catch (SQLException ex) {
            trataProblemaBancoDados(ex);
        }

    }

    /**
     * Fecha a conexão, o PreparedStatement e o ResultSet.
     *
     * @param con a conexão a ser fechada.
     * @param stmt o PreparedStatement a ser fechado.
     * @param rs o ResultSet a ser fechado.
     */
    public static void closeConnection(Connection con, PreparedStatement stmt, ResultSet rs){
        // fecha as conexoes anteriores
        closeConnection(con, stmt);
        try {
            // se existir fecha o ResultSet
            if(rs != null){
                rs.close();
            }
        } catch (SQLException ex) {
            trataProblemaBancoDados(ex);
        }

    }

    /**
     * Trata as execoes geradas pelo banco de dados
     *
     * @param ex a excecao gerada
     */
    public static void trataProblemaBancoDados(Exception ex){
        // cria uma tela de erro para mostrar que não conseguiu conectar ao banco
        Alert alert = new Alert(Alert.AlertType.ERROR);
        alert.setTitle("Erro: " + ex.getMessage());
        alert.setContentText("Por favor, verifique sua conexao com o banco de dados MySQL.");
        alert.showAndWait();
    }
}

File: trabalho_do_semestre_python/src/main/conexao/FabricaConexao.py
import mysql.connector
from tkinter import messagebox

class FabricaConexao:
    # Atributos estáticos para configurar a conexão com o banco de dados MySQL
    DRIVE = "mysql.connector"
    HOST = "localhost"
    PORT = 3306
    DATABASE = "produtos_bd"
    USER = "Adm"
    PASS = "Secret_123"

    @staticmethod
    def getConection():
        # Retorna uma conexão com o banco de dados MySQL usando os atributos de configuração
        try:
            return mysql.connector.connect(
                host=FabricaConexao.HOST,
                port=FabricaConexao.PORT,
                user=FabricaConexao.USER,
                password=FabricaConexao.PASS,
                database=FabricaConexao.DATABASE
            )
        except mysql.connector.Error as ex:
            raise RuntimeError(f"Erro na conexão: {ex}")

    @staticmethod
    def closeConnection(con):
        # Fecha a conexão com o banco de dados, se estiver aberta
        try:
            if con.is_connected():
                con.close()
        except mysql.connector.Error as ex:
            FabricaConexao.trataProblemaBancoDados(ex)

    @staticmethod
    def closeConnectionWithStmt(con, stmt):
        # Fecha a conexão com o statement do banco de dados
        FabricaConexao.closeConnection(con)
        try:
            if stmt:
                stmt.close()
        except mysql.connector.Error as ex:
            FabricaConexao.trataProblemaBancoDados(ex)

    @staticmethod
    def closeConnectionWithStmtAndRs(con, stmt, rs):
        # Fecha a conexão com o statement e o result set do banco de dados
        FabricaConexao.closeConnectionWithStmt(con, stmt)
        try:
            if rs:
                rs.close()
        except mysql.connector.Error as ex:
            FabricaConexao.trataProblemaBancoDados(ex)

    @staticmethod
    def trataProblemaBancoDados(ex):
        # Exibe uma mensagem de erro em uma caixa de diálogo, tratando problemas com o banco de dados
        messagebox.showerror("Erro", f"Erro: {ex.msg}\nPor favor, verifique sua conexão com o banco de dados MySQL.")
